src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+        roundedHourDifference,
+        userShiftNotes,
+      ).then(result => {
+        shiftDisplayComponent.loadShifts()
+        setSelectedDateOut(null);
+        setSelectedDateIn(null);
+        setUserShiftNotes('');
+        setUserJobs([]);
+        setHoursWorkedInDecimal(0);
+      }).catch(err => {
+        if (err.toLowerCase().includes('overlap')) {
+          setIsOverlappingShift(true);
+        }
+      });
+    };
+
+    const saveShift = async (
+      eml,
+      shiftStart,
+      shiftEnd,
+      hoursWorked,
+      shiftNotes,
+    ) => {
+      await jobs.saveShiftForUser(
+        eml,
+        shiftStart,
+        shiftEnd,
+        hoursWorked,
+        shiftNotes,
+      );
+    };
+
+    const jobsMenuItems = userJobs ? (
+      userJobs.map(job => (
+        <MenuItem label={job.POSTITLE} value={job} key={job.EMLID}>
+          {job.POSTITLE}
+        </MenuItem>
+      ))
+    ) : (
+        <></>
+      );
+
+    const isLeapYear = date => {
+      if (date.getFullYear() % 4 === 0) {
+        if (date.getFullYear() % 100 === 0) {
+          if (date.getFullYear() % 400 !== 0) {
+            return 'false';
+          }
+          if (date.getFullYear() % 400 === 0) {
+            return 'true';
+          }
+        }
+        if (date.getFullYear() % 100 !== 0) {
+          return 'true';
+        }
+      }
+      if (date.getFullYear() % 4 !== 0) {
+        return 'false';
+      }
+    };
+
+    const getNextDate = date => {
+      let is30DayMonth =
+        date.getMonth() === 3 ||
+        date.getMonth() === 5 ||
+        date.getMonth() === 8 ||
+        date.getMonth() === 10;
+
+      let isFebruary = date.getMonth() === 1;
+      let isDecember = date.getMonth() === 11;
+      let nextDate;
+      let monthToReturn;
+      let yearToReturn;
+
+      if (isFebruary) {
+        if (isLeapYear) {
+          if (date.getDate() === 29) {
+            nextDate = 1;
+            monthToReturn = 2;
+            yearToReturn = date.getFullYear();
+          } else {
+            nextDate = date.getDate() + 1;
+            monthToReturn = date.getMonth();
+            yearToReturn = date.getFullYear();
+          }
+        } else if (date.getDate() === 28) {
+          nextDate = 1;
+          monthToReturn = 2;
+          yearToReturn = date.getFullYear();
+        } else {
+          nextDate = date.getDate() + 1;
+          monthToReturn = date.getMonth();
+          yearToReturn = date.getFullYear();
+        }
+      } else if (isDecember) {
+        if (date.getDate() === 31) {
+          nextDate = 1;
+          monthToReturn = 0;
+          yearToReturn = date.getFullYear() + 1;
+        } else {
+          nextDate = date.getDate() + 1;
+          monthToReturn = date.getMonth();
+          yearToReturn = date.getFullYear();
+        }
+      } else if (is30DayMonth) {
+        if (date.getDate() === 30) {
+          nextDate = 1;
+          monthToReturn = (date.getMonth() + 1) % 12;
```

Comment:
Is the modulo 12 necessary if you handle the December case separately?

Reply:
You are probably right, we will test that